index.js

```diff
@@ -197,7 +197,11 @@ const onChange = (object, onChange, options = {}) => {
 				}
 
 				applyPath = pathCache.get(target);
-				applyPath = applyPath.slice(0, applyPath.lastIndexOf(PATH_SEPARATOR));
+				const firstPathPart = applyPath.lastIndexOf(PATH_SEPARATOR);
+				if (firstPathPart !== -1) {
+					applyPath = applyPath.slice(0, firstPathPart);
+				}
```

Comment:
This slice is here to remove the last item from the path, but your change doesn't remove the last item if there is only one item in the path.